bundles/org.openhab.core.model.core/src/main/java/org/openhab/core/model/core/internal/ModelRepositoryImpl.java

```diff
@@ -96,16 +100,18 @@ public EObject getModel(String name) {
     @Override
     public boolean addOrRefreshModel(String name, final InputStream originalInputStream) {
         Resource resource = null;
+        InputStream inputStream = null;
         try {
-            InputStream inputStream = null;
             if (originalInputStream != null) {
                 byte[] bytes = IOUtils.toByteArray(originalInputStream);
-                String validationResult = validateModel(name, new ByteArrayInputStream(bytes));
-                if (validationResult != null) {
-                    logger.warn("Configuration model '{}' has errors, therefore ignoring it: {}", name,
-                            validationResult);
-                    removeModel(name);
-                    return false;
+                try (InputStream validationStream = new ByteArrayInputStream(bytes)) {
```

Comment:
I'm OK with the changes, but this change will probably not fix memory leaks since the close method of a `ByteArrayInputStream` doesn't do anything:

```java
    /**
     * Closing a {@code ByteArrayInputStream} has no effect. The methods in
     * this class can be called after the stream has been closed without
     * generating an {@code IOException}.
     */
    public void close() throws IOException {
    }
```

It's just an `InputStream` backed by a byte array that doesn't hold any other resources besides the array and a few variables. It doesn't hurt to always use try-with-resources to prevent issues, but it's only really required for a few Stream implementations.